### ISSUE TITLE
left_joins performs INNER JOIN with ActiveRecord 6.1.4.4

### ISSUE THREAD

**Issue by yysaki** [rzane/baby_squeel#117]
#### Issue

I try to use baby_squeel 1.4.2.
`left_joins` should perform `LEFT OUTER JOIN`, but it performs `INNER JOIN` with ActiveRecord 6.1.4.4 and later.
It doesn't occur with ActiveRecord 6.0.4.4 and below.

#### Reproduction

```ruby
require 'bundler/inline'
require 'minitest/spec'
require 'minitest/autorun'

gemfile true do
  source 'https://rubygems.org'
  gem 'activerecord', '~> 6.1.4.4' # which Active Record version?
  gem 'sqlite3'
  gem 'baby_squeel', github: 'rzane/baby_squeel'
end

ActiveRecord::Base.establish_connection(adapter: 'sqlite3', database: ':memory:')

ActiveRecord::Schema.define do
  create_table :users, force: true do |t|
    t.string :name
  end
  create_table :dogs, force: true do |t|
    t.references(:user)
    t.string :name
  end
end

class User < ActiveRecord::Base
  has_many :dogs
end

class Dog < ActiveRecord::Base
  belongs_to :user
end

class BabySqueelTest < Minitest::Spec
  it 'works' do
    sql = User.left_joins(:dogs).to_sql
    sql.must_equal %{
      SELECT "users".* FROM "users"
      LEFT OUTER JOIN "dogs" ON "dogs"."user_id" = "users"."id"
    }.squish
  end
end
```

#### Test result

```
# Running:

DEPRECATED: global use of must_equal from issue.rb:35. Use _(obj).must_equal instead. This will fail in Minitest 6.
F

Finished in 0.030122s, 33.1983 runs/s, 33.1983 assertions/s.

  1) Failure:
BabySqueelTest#test_0001_works [issue.rb:35]:
--- expected
+++ actual
@@ -1 +1 @@
-"SELECT \"users\".* FROM \"users\" LEFT OUTER JOIN \"dogs\" ON \"dogs\".\"user_id\" = \"users\".\"id\""
+"SELECT \"users\".* FROM \"users\" INNER JOIN \"dogs\" ON \"dogs\".\"user_id\" = \"users\".\"id\""


1 runs, 1 assertions, 1 failures, 0 errors, 0 skips
```

**Comment by rzane** [comment]
Hey @rocket-turtle, do you have any idea what might be causing this issue? It looks like it might be a Ransack issue to me, since it's breaking Active Record itself.

**Comment by rocket-turtle** [comment]
Thank you for your great bug report @yysaki.
If you start a new project I would not recommend using baby_squeel.
The DSL and function is very cool and feels natural to use but the userbase seems small and implementation is a bit hacky.
It's sad that working with aliases in AR is so broken/difficult/whatever. 

@rzane This should fix the bug: https://github.com/rzane/baby_squeel/pull/118/files

**Comment by yysaki** [comment]
Hi @rzane, @rocket-turtle, thanks for your prompt support.
I'm now managing a rails 5.2 app that uses this gem and investigating how to upgrade rails version.
As far as I see, it works fine at 6.0, but I came across this bug at 6.1.
So I'm thinking about how to deal with this gem.

Also, I found a similar unexpected bug in test, so I will create another issue again.

**Comment by rzane** [comment]
I just released [v1.4.3](https://rubygems.org/gems/baby_squeel/versions/1.4.3), which includes a fix for this issue.
